### PR TITLE
Adds Emergency Plasma/Nitrogen Tanks to Medical Storage

### PR DESCRIPTION
**Pull request by FlattestGuitar**
:cl: FlattestGuitar
rscadd: Added emergency nitrogen and plasma tanks, you can find two of each in the medical storage
/:cl:

Also retouched quite a bit of the job equip code, because I'm a messy person like that.

### CODE REVIEW

File: code/game/jobs/job/engineering.dm [hunk 1]
```diff
@@ -32,11 +32,8 @@
 		H.equip_or_collect(new /obj/item/clothing/head/hardhat/white(H), slot_head)
 		H.equip_or_collect(new /obj/item/weapon/storage/belt/utility/full(H), slot_belt)
 		H.equip_or_collect(new /obj/item/clothing/gloves/color/black/ce(H), slot_gloves)
-		if(H.backbag == 1)
-			H.equip_or_collect(new /obj/item/weapon/storage/box/engineer(H), slot_r_hand)
-		else
-			H.equip_or_collect(new /obj/item/weapon/storage/box/engineer(H.back), slot_in_backpack)
-			H.equip_or_collect(new /obj/item/weapon/melee/classic_baton/telescopic(H.back), slot_in_backpack)
+		give_kit(H)
+		H.equip_or_collect(new /obj/item/weapon/melee/classic_baton/telescopic(H.back), slot_in_backpack)
```
Comment (FlattestGuitar) [inline code/game/jobs/job/engineering.dm]: Actually, I think I could get away with just switching to this method for every single item that's put into a backpack. Much cleaner.

**Comment by IcyV** [comment]
>IPCs get an emergency welder and a cable coil.

This doesn't seem entirely equivalent to the other species starting gear. Correct me if I'm wrong, since I ditch my box instantly more often than not, but the autoinjectors don't heal and just act as a painkiller. Not sure one species should be able to have a decent, free heal that happens to already be plentiful all over the station and is a useful tool in many situations.

I was also kind of thinking that it would be interesting to see nobody start with internals anymore. Everyone just gets a single crowbar or such. Everyone having a box of free supplies pretty much invalidates having any emergency lockers on the station save for rare cases.

**Comment by FlattestGuitar** [comment]
>IPC stuff

Well, they don't really have anything equivalent to epi or internals, so I gave them that. In hindsight, this was probably a bad idea, but I'll leave it be for now and see what other people have to say.

>I was also kind of thinking that it would be interesting to see nobody start with internals anymore.

That's actually a very interesting idea. I'd like to hear some more opinions on this.

**Comment by jughu** [comment]
@IcyV autoinjectors are for suffocation damage nerd it caps it at 25

**Comment by IcyV** [comment]
@FlattestGuitar Yeah, I wish I had something to suggest in place of the welder/cable but I don't honestly know what would fit. It just seems skewed is all. IPCs being immune to most things makes it tricky for this kind of situation.

@jughu Thanks. Never really needed one since, y'know, starting with internals.

**Comment by Crazylemon64** [comment]
@IcyV The autoinjector's for if, like, your coworker gets floored by an explosion and is in deep crit - with the autoinjector, they stand a good chance of pulling through, instead of succumbing. The situations where it would be useful for yourself, you'd probably be incapable of actually using it.

**Comment by Fullofskittles** [comment]
> Everyone having a box of free supplies pretty much invalidates having any emergency lockers on the station save for rare cases.

I agree. I'd like to see things be more risky and more dangerous instead of adding more safety nets. If someone wants to carry around stuff like that then they should easily be able to obtain those things normally via interaction with the rest of the crew instead of just starting with it.

I cant really think of many instances as a regular vox player that my actual tank has legit been missing and i needed a replacement. Maybe two or three times in the few years I've played here and it certainly gave atmos something to do.

> I was also kind of thinking that it would be interesting to see nobody start with internals anymore.

> That's actually a very interesting idea. I'd like to hear some more opinions on this.

I'd totally be up for this. If there's one thing I miss in this game it's the actual feeling of danger. There's also so many emergency lockers with internals around the station that are utterly useless because everyone already has those things.

**Comment by LittleBigKid2000** [comment]
What if IPCs were given a small emergency nanopaste sprayer. Single use, heals a bit of brute and burn, deals a bit of brain damage because bluespace suchandsuch.

**Comment by jughu** [comment]
Give them a nanites salve

**Comment by Fox-McCloud** [comment]
- Vox don't need emergency oxygen tanks; they already technically have one, not to mention that they start out with a full sized tank that lasts the entire shift
- Plasmamen are in the same boat as Vox

- Hell freaking no on the welder and cable coil for IPCs--it's effectively giving them a bruise pack and ointment. Auto-injectors are very niche use and *largely used for other crewmembers not yourself*; given that IPCs are pain immune, this is far more effective for them than an auto-injector could ever hope to be.

**Comment by Lady-Luck** [comment]
I do feel like something should be done with the roundstart box, but this doesn't feel right to me for the reasons Fox stated.

Giving IPCs a small vial of liquid solder instead of oxygen/mask would be a neat idea.

It would also be nice if the roundstart box had its own sprite.

**Comment by FalseIncarnate** [comment]
As an IPC player, I normally don't use my roundstart supplies, but I hold on to them for the precise reason people point out the inclusion of epi-pens: to give to at-risk crew. Especially since I play as an atmos tech, I tend to find myself in dangerous areas, so having an epi-pen and set of internals handy means I can potentially save the life of a crew member in a depressurized zone that might not be able to help themselves.

While the concept of a species-related starter box is neat, it is largely unnecessary since the ones who ultimately could benefit from it already are in a position where they'd rarely need spares. This only would really help if someone stole the nitrogen or plasma tank from a vox or plasmaman and they needed an emergency tank so they could get to atmos for a new one. And that is such a rare occurance that it hardly seems sensible to make this change.

If anything, toss an emergency nitrogen or plasma tank in the emergency O2 lockers, not their boxes.

**Comment by Lady-Luck** [comment]
Wait, what does this PR do now

**Comment by FlattestGuitar** [comment]
Empty PR, everyone go home.

It's in the description, though I'm not sure where this is going.

File: code/game/objects/items/weapons/tanks/tank_types.dm [hunk 2]
```diff
@@ -202,9 +202,30 @@
 	item_state = "emergency_vox"
 	volume = 25
 
+	New()
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/tanks/tank_types.dm]: relative pathing.

File: code/game/objects/items/weapons/tanks/tank_types.dm [hunk 3]
```diff
@@ -202,9 +202,30 @@
 	item_state = "emergency_vox"
 	volume = 25
 
+	New()
+		..()
+		air_contents.oxygen -= (3*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)
+		air_contents.nitrogen = (10*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)
+		return
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/tanks/tank_types.dm]: Trailing return

File: code/game/objects/items/weapons/tanks/tank_types.dm [hunk 4]
```diff
@@ -202,9 +202,30 @@
...
+		air_contents.nitrogen = (10*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)
+		return
+
+/obj/item/weapon/tank/emergency_oxygen/nitrogen
+	name = "emergency nitrogen tank"
+	desc = "An emergency tank designed specifically for Vox."
+	icon_state = "emergency_nitrogen"
+	volume = 3
+
+	New()
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/tanks/tank_types.dm]: relative pathing.

File: code/game/objects/items/weapons/tanks/tank_types.dm [hunk 5]
```diff
@@ -202,9 +202,30 @@
...
+		air_contents.oxygen -= (3*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)
+		air_contents.nitrogen = (10*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)
+
+/obj/item/weapon/tank/emergency_oxygen/plasma
+	name = "emergency plasma tank"
+	desc = "An emergency tank designed specifically for Plasmamen."
+	icon_state = "emergency_p"
+	volume = 3
 
 	New()
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/tanks/tank_types.dm]: relative pathing.

**Comment by Lady-Luck** [comment]
@FlattestGuitar I'll give you a dollar if you give the roundstart box a new sprite.

**Comment by FlattestGuitar** [comment]
I don't even know what this PR does anymore.

Ideas:
- Do we add plasma and nitrogen emergency tanks to EVA and/or oxy lockers?
- Is there anything related that we could improve?
- How would the new internals box look? I kinda dislike how the sprite is the same as a normal box, but I don't know how to improve it.

**Comment by Lady-Luck** [comment]
spare plasma and nitrogen tanks should probably be in Medical's secondary storage.

**Comment by IcyV** [comment]
Don't think I'm a fan of those sprites. The colors stand out really hard and I wouldn't have any kind of guess on which could be which at first sight.

**Comment by Lady-Luck** [comment]
>Don't think I'm a fan of those sprites. The colors stand out really hard and I wouldn't have any kind of guess on which could be which at first sight.

Uhm, those aren't new sprites. Those are the ones those tanks have always had.

**Comment by Lady-Luck** [comment]
>Don't think I'm a fan of those sprites. The colors stand out really hard and I wouldn't have any kind of guess on which could be which at first sight.

Plasma has traditionally been purple, and vox tanks have traditionally been reddish-orange. I don't see the problem.

File: code/game/jobs/job/civilian.dm [hunk 6]
```diff
@@ -18,14 +18,11 @@
 		if(4) H.equip_or_collect(new /obj/item/weapon/storage/backpack/satchel(H), slot_back)
 	H.equip_or_collect(new /obj/item/clothing/under/color/random(H), slot_w_uniform)
 	H.equip_or_collect(new /obj/item/clothing/shoes/black(H), slot_shoes)
-	if(H.backbag == 1)
-		H.equip_or_collect(new /obj/item/weapon/storage/box/survival(H), slot_r_hand)
-	else
-		H.equip_or_collect(new /obj/item/weapon/storage/box/survival(H.back), slot_in_backpack)
```
Comment (Fox-McCloud) [inline code/game/jobs/job/civilian.dm]: you've removed a lot of instances of `H.back`, but other times you've left it in..why?

**Comment by Fox-McCloud** [comment]
That's not..plasma purple though, that's like...pink.

I'm still not sure the necessity of emergency nitrogen/plasma tanks? I really don't know
 
@TheDZD

**Comment by IcyV** [comment]
>Plasma has traditionally been purple, and vox tanks have traditionally been reddish-orange. I don't see the problem.

Plasma
Plasma has traditionally come in an orange canister or tank. Even if the tank is to represent the gas itself, it's the wrong color. 

N2O
N20 has traditionally come as a darker red with a white stripe. Some forms have something at least close to a white stripe.

There's no reason to break an established and consistent theme.

**Comment by Lady-Luck** [comment]
>I'm still not sure the necessity of emergency nitrogen/plasma tanks? I really don't know

There's been situations where vox have been incapacitated in medbay in cryotubes for quite a while because of their tanks being missing. It would be very nice for medbay to have a couple spares in secondary storage.

**Comment by Faryaa** [comment]
Some kind of emergency nano-salve for ipcs would be neat. Having O2 lockers more useful would be neat as well. Maybe remove internals from boxes all together and give crew emergency aid kits? Tiny red sealed plastic bags filled with very small amount of medicine. They would contain: 1 charcoal pill, 1 emergency injector, 1 vial bactericyde salve(prevents wounds from getting infection), 1 emergency bandage(stops bleeding, 1 use) and 1 emergency flare. The bag itself cannot be refilled with anything but you can keep it in your pocket.
ipc would have emergency repair kit. Orange sealed bag with liquid soldier, 1 flare and emergency nanopaste tube. 
Drask would have blue kit similar to human one, but with something which coold cool the body down for short time.
Vox would probably have no injector but more charcoal pills.

**Comment by FlattestGuitar** [comment]
Here are the new tanks, I recolored both of them.

Also adressed some of the stuff Fox mentioned.

As for the survival kits, since this is no longer a PR that directly involves them, I think we should take the conversation to the forums.

**Comment by Lady-Luck** [comment]
They look great

**Comment by Faryaa** [comment]
http://80.244.78.90/phpBB3/viewtopic.php?f=12&t=8520&sid=8208df918ce541c93d9a3acdca604e8d
This thread is not rotten. So feel free to discuss here.

**Comment by IcyV** [comment]
Looks much better and actually identifiable now, good work. I'd have gone with a barely darker shade of orange for the plasma that more matched the canister, but that's purely my personal taste at this point.

**Comment by Fox-McCloud** [comment]
@IcyV 

Keep in mind There's a difference between Nitrogen (what Vox breathe and what's in standard station air) and N2O (laughing gas/sleeping gas).

N2O has the stripe, but nitrogen does not.

**Comment by IcyV** [comment]
@Fox-McCloud The two tanks next to the canister in the picture I put up are what Vox can start with. Nitro tanks that have a stripe. It's very much odd that it's the same style as an N20 canister, but that happens to be how it's done for some reason.

**Comment by FlattestGuitar** [comment]
The nitrogen tank was already in the files, I only added the stripe to it, while the plasma tank uses copypasted colors of the larger canister.

I think the current palette works perfectly.

Personally, I find it a good idea to add them to medbay storage. We could possibly clean out EVA of the Vox gear, which is there for *reasons*.

**Comment by Fox-McCloud** [comment]
>Personally, I find it a good idea to add them to medbay storage. We could possibly clean out EVA of the Vox gear, which is there for reasons.

Vox need to not have their own :snowflake: space suit and it just uses the regular space suit with species sprite adjustments....like how hardsuits work.

**Comment by jughu** [comment]
I liked modification kits tbh

**Comment by FlattestGuitar** [comment]
Well, considering Vox can wear all normal space suits just fine, should I just remove the ones in EVA and replace them with just tanks of plasma and nitrogen? Seems to me like it would be a good alternative to the very small emergency tanks you can find at medbay.

**Comment by jughu** [comment]
Why would a fox need a space suit anyways a soft suit that is

**Comment by Aurorablade** [comment]
@jughu ....I don't see how this relvent to anything...

File: code/game/jobs/job/civilian.dm (hunk 6 above)
Comment (Fox-McCloud) [inline code/game/jobs/job/civilian.dm]: Still need a reason here =p

**Comment by Taaketa** [comment]
Sinve when did Vox breath N20?

File: code/game/jobs/job/civilian.dm (hunk 6 above)
Comment (FlattestGuitar) [inline code/game/jobs/job/civilian.dm]: I've removed all instances of `H.back` where I could. It doesn't really matter where you spawn the item if you spawn it for the sole purpose of equipping it to a specific slot.

`equip_or_collect()` without any additional backpack checks does that more cleanly. What I did just makes it all more readable.

Comment (Fox-McCloud) [inline code/game/jobs/job/civilian.dm]: Yes, I know, I just don't get why you left some `H.back`s in, and axed others =p 

If you're going to kill some off..kill them all off, please.

Comment (FlattestGuitar) [inline code/game/jobs/job/civilian.dm]: My GitHub did something very weird. Apparently it hadn't pushed the commit that adressed that properly. Now it's all good...without a new commit. :eyes:

**Comment by FalseIncarnate** [comment]
The PR no longer adjusts survival kits, so you might want to adjust the title to reflect that. The closest it comes is refactoring the starting equipment's handling of the survival kit, which honestly could just be said as "refactors spawn equipment" or something to clarify that the survival kits aren't being touched themselves.

**Comment by Plotron** [comment]
Another upside of adding these emergency 'exotic air' tanks to Medbay - asshole medical may now use plasma tanks to poison people.
:+1:

**Comment by Fox-McCloud** [comment]
@TheDZD said to drop the ones from EVA storage, but leave in the ones in medical.

Will merge, once that's done.

**Comment by Plotron** [comment]
Yeah, ABSOLUTELY no need to keep such tanks in EVA Storage.

**Comment by FlattestGuitar** [comment]
Should I take the Vox suits out as well then or is that for a different PR?

**Comment by FlattestGuitar** [comment]
Welp, I reset instead of reverting, so the commit isn't there, but EVA is now back to normal with the skrexxy Vox suits.

Considering this PR is already as spaghetti as it gets, I'd rather their removal was done in a different one.

I guess this one's ready for merge.

**Pull request merged**